subsys/net/lib/sockets/sockets.c

```diff
@@ -2590,6 +2592,46 @@ int z_vrfy_zsock_getsockopt(int sock, int level, int optname,
 #include <syscalls/zsock_getsockopt_mrsh.c>
 #endif /* CONFIG_USERSPACE */
 
+static int ipv4_multicast_group(struct net_context *ctx, const void *optval,
```

Comment:
The `ctx` argument is unused, maybe remove it from this static function?

Reply:
Actually I think we can leave the ctx there as it can be used to select an interface if user has not supplied one. I will add support for this in next version.

Reply:
Ok!